hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -178,6 +178,9 @@ extends Importer:
   
   def term(tree: Tree, inAppPrefix: Bool = false): Ctxl[Term] =
   trace[Term](s"Elab term ${tree.showDbg}", r => s"~> $r"):
+    def maybeModuleMethod(t: Term): Ctxl[Term] =
+      if !inAppPrefix && ModuleChecker.isModuleMethodApp(t) then moduleMethod(t)
```

Comment:
Why `!inAppPrefix`?

Admittedly we should document what the purpose of `inAppPrefix` does: it's used to avoid generating `undefined`-checks when a field being accessed is actually called like a method, because in this case `undefined` will crash as needed already.

I don't think this has anything to do with module methods.

Reply:
I couldn't think of a way to have module methods partially applied. If we do that, the result may not be recognized as a module method and then cannot be further elaborated with type information. I remember we discussed this, and we ended up with to reject partial application on module methods. (not implemented yet)

Reply:
If the omitted parameter lists don't have any special feature in them (no `using`, etc.) then there's no reason to forbid it, as it's functionally equivalent to a module method returning a lambda. There would be no reason to reject that.

> I remember we discussed this, and we ended up with to reject partial application on module methods.

IIRC I was just suggesting to have this arbitrary implementation restriction to ease progress on the first PR.

Reply:
Alright. Let's formally restrict partial application if the resulting function does require further elaboration with type information. Otherwise, let it behaves just like normal methods.

Reply:
> if the resulting function does require further elaboration with type information

Note that we should also check that the result type also does not need module-method treatment. E.g., it shouldn't return a module!